extract_learning_curves raises ValueError

https://github.com/rlai-lab/rl-evaluation/blob/7483d5ccbbea2015726255f66c625dcd109257f5/rlevaluation/temporal.py#L64

```
  File "/home/steven/Github/continual-foragax-agents/.venv/lib/python3.12/site-packages/rlevaluation/temporal.py", line 64, in extract_learning_curves
    if idx:
       ^^^
ValueError: The truth value of an empty array is ambiguous. Use `array.size > 0` to check that an array is not empty.
```

```
x
[    0   500  1000  1500  2000  2500  3000  3500  4000  4500  5000  5500
  6000  6500  7000  7500  8000  8500  9000  9500 10000 10500 11000 11500
 12000 12500 13000 13500 14000 14500 15000 15500 16000 16500 17000 17500
 18000 18500 19000 19500 20000 20500 21000 21500 22000 22500 23000 23500
 24000 24500 25000 25500 26000 26500 27000 27500 28000 28500 29000 29500
 30000 30500 31000 31500 32000 32500 33000 33500 34000 34500 35000 35500
 36000 36500 37000 37500 38000 38500 39000 39500 40000 40500 41000 41500
 42000 42500 43000 43500 44000 44500 45000 45500 46000 46500 47000 47500
 48000 48500 49000 49500 50000 50500 51000 51500 52000 52500 53000 53500
 54000 54500 55000 55500 56000 56500 57000 57500 58000 58500 59000 59500
 60000 60500 61000 61500 62000 62500 63000 63500 64000 64500 65000 65500
 66000 66500 67000 67500 68000 68500 69000 69500 70000 70500 71000 71500
 72000 72500 73000 73500 74000 74500 75000 75500 76000 76500 77000 77500
 78000 78500 79000 79500 80000 80500 81000 81500 82000 82500 83000 83500
 84000 84500 85000 85500 86000 86500 87000 87500 88000 88500 89000 89500
 90000 90500 91000 91500 92000 92500 93000 93500 94000 94500 95000 95500
 96000 96500 97000 97500 98000 98500 99000 99500]
```

It seems like this fails an existing test.

```
tests/test_temporal.py F                                                                         [100%]

=============================================== FAILURES ===============================================
____________________________________ test_extract_learning_curves1 _____________________________________

    def test_extract_learning_curves1():
        df = learning_curve_data()
    
>       got_x, got_y = extract_learning_curves(
            df,
            { 'alpha': 0.01, 'beta': 4, 'gamma': 0.99 },
            metric='metric',
            data_definition=DataDefinition(
                hyper_cols=['alpha', 'beta', 'gamma'],
                seed_col='seed',
                time_col='time',
            )
        )

tests/test_temporal.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

df = shape: (6_750, 6)
┌───────┬──────┬───────┬──────┬───────────┬──────┐
│ alpha ┆ beta ┆ gamma ┆ time ┆ metric    ┆ seed ...3.58681  ┆ 9    │
│ 0.001 ┆ 4    ┆ 0.9   ┆ 989  ┆ 13.219681 ┆ 9    │
└───────┴──────┴───────┴──────┴───────────┴──────┘
hyper_vals = {'alpha': 0.01, 'beta': 4, 'gamma': 0.99}, metric = 'metric'
data_definition = DataDefinition(hyper_cols=['alpha', 'beta', 'gamma'], seed_col='seed', time_col='time', environment_col=None, algorithm_col=None)
interpolation = None

    def extract_learning_curves(
        df: pl.DataFrame,
        hyper_vals: dict[str, Any],
        metric: str,
        data_definition: DataDefinition | None = None,
        interpolation: Interpolation | None = None,
    ):
        """
        Takes a dataframe of shape:
          stepsize  seed  metric  time
          0.01      0     0.1     1
          0.01      0     0.2     2
          0.01      0     0.3     3
    
        and extracts a learning curve per seed for the given hyperparameters,
        where a "learning curve" is defined by a numpy array for x
        values and another y values.
        """
        dd = maybe_global(data_definition)
        cols = set(dd.hyper_cols).intersection(df.columns)
    
        sub = df.filter(**du.subset(hyper_vals, cols))
        groups = sub.group_by(dd.seed_col)
    
        xs: list[np.ndarray] = []
        ys: list[np.ndarray] = []
        for _, group in groups:
            non_na = group.drop_nulls(subset=metric)
            x = non_na[dd.time_col].to_numpy().astype(np.int64)
            y = non_na[metric].to_numpy().astype(np.float64)
    
            idx = np.argwhere(x[1:] <= x[:-1])
    
>           if idx:
               ^^^
E           ValueError: The truth value of an empty array is ambiguous. Use `array.size > 0` to check that an array is not empty.

rlevaluation/temporal.py:64: ValueError
```

--- a/rlevaluation/temporal.py
+++ b/rlevaluation/temporal.py
@@ -61,7 +61,7 @@ def extract_learning_curves(
 
         idx = np.argwhere(x[1:] <= x[:-1])
 
-        if idx:
+        if idx.size > 0:
             x = x[:idx[0][0]]
             y = y[:idx[0][0]]
 

